### PR TITLE
Astroport Withdrawer combined

### PR DESCRIPTION
**Pull request by keyleu**
Much simpler than LPer, combines the liquidity withdrawer in one contract.

### CODE REVIEW

File: contracts/services/astroport-withdrawer/src/contract.rs [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+                ));
+            }
+            let config = CONFIG.load(deps.storage)?;
+            actions::process_action(deps, env, info, action_msgs, config)
+        }
+        ExecuteMsg::UpdateConfig { new_config } => {
+            cw_ownable::assert_owner(deps.as_ref().storage, &info.sender)?;
+            let config = new_config.validate(deps.as_ref())?;
+            CONFIG.save(deps.storage, &config)?;
+            Ok(Response::new().add_attribute("method", "update_config"))
```
Comment (Art3miX) [inline contracts/services/astroport-withdrawer/src/contract.rs:57]: Same config thing

**Review by Art3miX — CHANGES_REQUESTED** [review]
Some comments from previous PRs apply here as well, didn't want to duplicate them.

File: contracts/services/astroport-withdrawer/src/contract.rs (hunk 1 above)
Comment (keyleu) [reply contracts/services/astroport-withdrawer/src/contract.rs]: Refactored contract

File: contracts/services/astroport-withdrawer/src/astroport_cw20.rs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        cfg.pool_addr.clone(),
+        &valence_astroport_utils::astroport_cw20_lp_token::PoolQueryMsg::Share {
+            amount: balance_response.balance,
+        },
+    )?;
+
+    // Create the withdraw and send messages
+    let withdraw_msg = CosmosMsg::Wasm(cosmwasm_std::WasmMsg::Execute {
+        contract_addr: token_addr.to_string(),
+        msg: to_json_binary(&Cw20ExecuteMsg::Send {
```
Comment (stiiifff) [inline contracts/services/astroport-withdrawer/src/astroport_cw20.rs:48]: I don't understand this part .. can you explain how it works ?

Comment (keyleu) [reply contracts/services/astroport-withdrawer/src/astroport_cw20.rs:48]: Yes, in Cw20 we have both Transfer and Send.
- Transfer is similar to a bank send message, just send the CW20 token to an address.
- Send on the other hand, is to send CW20 tokens to a contract and atomically trigger a hook on that contract (in this case, the withdrawliquidity message that astroport defines in its pool contract that I specify below)

In summary, what I'm saying is: Send these tokens (they are cw20, so it's a smart contract execution) to this contract address and immediately trigger the action (hook) that I'm specifying here. Astroport will receive those cw20 tokens and do the action in the contract.

If it's still not clear lmk and we can hop on a quick call.

Comment (stiiifff) [reply contracts/services/astroport-withdrawer/src/astroport_cw20.rs:48]: Understood, all clear, thanks !

**Review by stiiifff — APPROVED** [review]

File: contracts/services/astroport-withdrawer/README.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+## Configuration
+
+The service is configured on instantiation via the `ServiceConfig` type.
+
+```rust
+pub struct ServiceConfig {
+    // Account from which the funds are LPed
+    pub input_addr: String,
+    // Account to which the LP tokens are forwarded
+    pub output_addr: String,
```
Comment (Art3miX) [inline contracts/services/astroport-withdrawer/README.md:40]: Use `ServiceAccountType` here

**Review by Art3miX — APPROVED** [review]

**Pull request merged**